### PR TITLE
added possibility to send custom cas

### PR DESCRIPTION
**Pull request by dominikkukacka**
<a href='#crh-start'></a><a href='#crh-data-%7B%22approved%22%3A%20%7B%22https%3A//github.com/TheBay0r%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8627051%3Fv%3D3%22%7D%7D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Jumio/con-rest/pull/64%23issuecomment-70809478%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222015-01-21T09%3A42%3A57Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8627051%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TheBay0r%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%207150a049b2124717d4d620731ae4c181ec666aa2%20server/api.js%2016%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Jumio/con-rest/pull/64%23discussion_r23299664%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22In%20the%20future%20we%20could%20think%20of%20setting%20%5C%22strictSSL%3A%20false%5C%22%20when%20certificates%20is%20null%20or%20undefined%22%2C%20%22created_at%22%3A%20%222015-01-21T13%3A49%3A42Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8627051%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TheBay0r%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20server/api.js%3AL253-263%22%7D%2C%20%22Pull%2029604c3585568b838754f11e2bfbcbdec7220062%20config.json%207%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Jumio/con-rest/pull/64%23discussion_r23283674%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Please%20add%20default%20vars%2C%20these%20are%20no%20usual%20default%20vars.%22%2C%20%22created_at%22%3A%20%222015-01-21T08%3A14%3A41Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1508400%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/EnoF%22%7D%7D%2C%20%7B%22body%22%3A%20%22should%20i%20add%20the%20%60certificates/test0x.pem%60%20to%20the%20repo%3F%20i%20don%27t%20think%20this%20is%20good.%22%2C%20%22created_at%22%3A%20%222015-01-21T08%3A55%3A24Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/243319%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dominikkukacka%22%7D%7D%2C%20%7B%22body%22%3A%20%22definately%20not%2C%20this%20is%20jumio%20related%2C%20not%20conREST%20general%20public%22%2C%20%22created_at%22%3A%20%222015-01-21T09%3A01%3A19Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1508400%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/EnoF%22%7D%7D%2C%20%7B%22body%22%3A%20%22yes%2C%20ok%20then%20i%20will%20change%20the%20path%20to%20something%20generic.%22%2C%20%22created_at%22%3A%20%222015-01-21T09%3A02%3A43Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/243319%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dominikkukacka%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20config.json%3AL23-32%22%7D%2C%20%22Pull%2029604c3585568b838754f11e2bfbcbdec7220062%20server/api.js%2015%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Jumio/con-rest/pull/64%23discussion_r23283703%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22What%20happens%20on%20localhost%3F%22%2C%20%22created_at%22%3A%20%222015-01-21T08%3A15%3A29Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1508400%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/EnoF%22%7D%7D%2C%20%7B%22body%22%3A%20%22it%20trys%20to%20load%20the%20specified%20certificates.%20if%20they%20are%20not%20found%20then%20no%20certificate%20%28empty%20string%29%20is%20sent.%22%2C%20%22created_at%22%3A%20%222015-01-21T08%3A57%3A16Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/243319%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dominikkukacka%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20server/api.js%3AL253-263%22%7D%7D%2C%20%22processed%22%3A%20%5B%22https%3A//github.com/Jumio/con-rest/pull/64%23issuecomment-70809478%22%2C%20%22https%3A//github.com/Jumio/con-rest/pull/64%23discussion_r23283674%22%2C%20%22https%3A//github.com/Jumio/con-rest/pull/64%23discussion_r23283703%22%2C%20%22https%3A//github.com/Jumio/con-rest/pull/64%23discussion_r23285262%22%2C%20%22https%3A//github.com/Jumio/con-rest/pull/64%23discussion_r23285356%22%2C%20%22https%3A//github.com/Jumio/con-rest/pull/64%23discussion_r23285586%22%2C%20%22https%3A//github.com/Jumio/con-rest/pull/64%23discussion_r23285655%22%2C%20%22https%3A//github.com/Jumio/con-rest/pull/64%23discussion_r23299664%22%5D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/TheBay0r'></a>

- [ ] <a href='#crh-comment-Pull 29604c3585568b838754f11e2bfbcbdec7220062 config.json 7'></a> &nbsp;<b><a href='https://github.com/Jumio/con-rest/pull/64#discussion_r23283674'>File: config.json:L23-32</a></b>
- <a href='https://github.com/EnoF'></a> Please add default vars, these are no usual default vars.
- <a href='https://github.com/dominikkukacka'></a> should i add the `certificates/test0x.pem` to the repo? i don't think this is good.
- <a href='https://github.com/EnoF'></a> definately not, this is jumio related, not conREST general public
- <a href='https://github.com/dominikkukacka'></a> yes, ok then i will change the path to something generic.
- [ ] <a href='#crh-comment-Pull 29604c3585568b838754f11e2bfbcbdec7220062 server/api.js 15'></a> &nbsp;<b><a href='https://github.com/Jumio/con-rest/pull/64#discussion_r23283703'>File: server/api.js:L253-263</a></b>
- <a href='https://github.com/EnoF'></a> What happens on localhost?
- <a href='https://github.com/dominikkukacka'></a> it trys to load the specified certificates. if they are not found then no certificate (empty string) is sent.
- [ ] <a href='#crh-comment-Pull 7150a049b2124717d4d620731ae4c181ec666aa2 server/api.js 16'></a> &nbsp;<b><a href='https://github.com/Jumio/con-rest/pull/64#discussion_r23299664'>File: server/api.js:L253-263</a></b>
- <a href='https://github.com/TheBay0r'></a> In the future we could think of setting "strictSSL: false" when certificates is null or undefined

<a href='https://www.codereviewhub.com/Jumio/con-rest/pull/64?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/Jumio/con-rest/pull/64?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/Jumio/con-rest/pull/64?approve=0'></a>&nbsp;<a href='https://github.com/Jumio/con-rest/pull/64'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by EnoF** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/Jumio/con-rest/pull/64'>Refresh</a>.<a href='#crh-update-none'></a>

File: config.json [hunk 1]
```diff
@@ -23,5 +23,9 @@
       "user": "myUserName",
       "pass": "myPassword"
     }
-  }
+  },
+  "certificates": [
+    "certificates/test0x.pem",
```
Comment (EnoF) [inline config.json]: Please add default vars, these are no usual default vars.

File: server/api.js [hunk 2]
```diff
@@ -253,7 +253,10 @@
     var options = {
       method: apiCall.method,
       url: apiCall.url,
-      headers: headers
+      headers: headers,
+      agentOptions: {
```
Comment (EnoF) [inline server/api.js]: What happens on localhost?

File: config.json (hunk 1 above)
Comment (dominikkukacka) [inline config.json]: should i add the `certificates/test0x.pem` to the repo? i don't think this is good.

File: server/api.js (hunk 2 above)
Comment (dominikkukacka) [inline server/api.js]: it trys to load the specified certificates. if they are not found then no certificate (empty string) is sent.

File: config.json (hunk 1 above)
Comment (EnoF) [inline config.json]: definately not, this is jumio related, not conREST general public

Comment (dominikkukacka) [inline config.json]: yes, ok then i will change the path to something generic.

**Comment by TheBay0r** [comment]
:shipit:

File: server/api.js [hunk 3]
```diff
@@ -253,7 +253,10 @@
...
+        ca: config.getCertificates()
```
Comment (TheBay0r) [inline server/api.js]: In the future we could think of setting "strictSSL: false" when certificates is null or undefined

**Pull request merged**